backend-api/template.yaml

```diff
@@ -243,6 +243,95 @@ Resources:
 
+  PrivateApiDomainName:
+    Type: AWS::ApiGateway::DomainName
+    Properties:
+      DomainName: !Sub
+        - private-${AWS::StackName}.${DNS_RECORD}
+        - DNS_RECORD: !FindInMap
+            - DNS
+            - !Ref Environment
+            - BaseDns
+      EndpointConfiguration:
+        Types:
+          - REGIONAL
+      RegionalCertificateArn: !Sub '{{resolve:ssm:/${Environment}/Platform/ACM/AsyncPrimaryZoneWildcardCertificateARN}}' # Uses a wildcard (in dev only?)
+      SecurityPolicy: TLS_1_2
+
+  PrivateApiBasePathMapping:
+    Type: AWS::ApiGateway::BasePathMapping
+    DependsOn: PrivateApiDomainName
+    Properties:
+      DomainName: !Sub
+        - private-${AWS::StackName}.${DNS_RECORD}
+        - DNS_RECORD: !FindInMap
+            - DNS
+            - !Ref Environment
+            - BaseDns
+      RestApiId: !Ref PrivateApi
+      Stage: !Ref PrivateApi.Stage
+
+  PrivateApiRecordSet:
+    Type: AWS::Route53::RecordSet
+    Properties:
+      Name: !Sub
+        - private-${AWS::StackName}.${DNS_RECORD}
+        - DNS_RECORD: !FindInMap
+            - DNS
+            - !Ref Environment
+            - BaseDns
+      Type: CNAME
+      HostedZoneId: !Sub '{{resolve:ssm:/${Environment}/Platform/Route53/PrimaryZoneID}}'
+      TTL: 60
+      ResourceRecords:
+        - !GetAtt PrivateApiLoadBalancer.DNSName # this will be the load balancer domain
+
+  PrivateApiLoadBalancer:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      IpAddressType: ipv4
+      Scheme: internal
+      Subnets:
+        - !ImportValue devplatform-vpc-PrivateSubnetIdA
+        - !ImportValue devplatform-vpc-PrivateSubnetIdB
+        - !ImportValue devplatform-vpc-PrivateSubnetIdC
+      Type: network
+
+  PrivateApiListener:
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    Properties:
+      DefaultActions:
+        - Order: 1
+          TargetGroupArn: !Ref PrivateApiTargetGroup
+          Type: forward
+      LoadBalancerArn: !Ref PrivateApiLoadBalancer
+      Port: 443
+      Certificates:
+        - CertificateArn: !Sub '{{resolve:ssm:/${Environment}/Platform/ACM/AsyncPrimaryZoneWildcardCertificateARN}}'
+      Protocol: TLS
+
+  PrivateApiTargetGroup:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    Properties:
+      HealthCheckEnabled: true
+      HealthCheckIntervalSeconds: 10
+      HealthCheckPort: 443
+      HealthCheckProtocol: TCP
+      HealthCheckTimeoutSeconds: 10
+      HealthyThresholdCount: 3
+      UnhealthyThresholdCount: 3
+      Port: 443
+      Protocol: TLS
+      Targets:
+        - Id: 10.0.10.23 # subnet-0d43f751301129abd (devplatform-vpc-PrivateSubnetA) eu-west-2c (euw2-az1)
+          Port: 443
+        - Id: 10.0.13.96 # subnet-094e66815780c4556 (devplatform-vpc-PrivateSubnetB) eu-west-2a (euw2-az2)
+          Port: 443
+        - Id: 10.0.14.16 # subnet-0d90ca1c3b3a83028 (devplatform-vpc-PrivateSubnetC) eu-west-2b (euw2-az3)
+          Port: 443
```

Comment:
where do these hard coded values come from? are they stable?

Reply:
The values are currently the IP addresses for the ENIs of the Api Gateway VPC Endpoint. I need to add the values for the other environments. 

https://docs.aws.amazon.com/vpc/latest/privatelink/privatelink-access-aws-services.html#aws-service-subnets-zones

> The IP addresses of an endpoint network interface will not change during the lifetime of its VPC endpoint.

We can be confident these values will persist, but they would need to be updated if we moved VPCs / recreated the devplatform VPC.

Reply:
is there any way of getting these dynamically, perhaps by having the VPC stack write them out to CF exports or something like that?